services/src/main/java/org/fao/geonet/api/GlobalExceptionController.java

```diff
@@ -318,6 +346,18 @@ public ApiError unsatisfiedParameterHandler(final Exception exception) {
         MissingResourceException.class
     })
     public ApiError missingResourceHandler(final Exception exception) {
+        storeException(exception);
+
         return new ApiError("missing_resource_parameter", exception);
     }
+
+    /**
+     * Stores the exception in the request attributes to be processed by the TransactionManager and
+     * rollback the transaction.
+     *
+     * @param exception
+     */
+    private void storeException(final Exception exception) {
```

Comment:
`storeException` is too generic, in this case you are specifically storing an the cause of an API exception which you intend to report to the user as an HTTP error result.

Recommend `storeApiErrorCause( Exception exception )`